### PR TITLE
Removed redundant template in '__delete_node()' member function of '__forward_list_base' and '__list_imp' classes.

### PR DESCRIPTION
**Pull request by 2LoS**

### CODE REVIEW

File: libcxx/include/forward_list [hunk 1]
```diff
@@ -555,7 +555,6 @@ protected:
     return __guard.__release_ptr();
   }
 
-  template <class... _Args>
```
[the remark this answers is not in the document]
Comment (philnik777) [reply libcxx/include/forward_list:558]: Looks like this is from #65614, which looks like it's just copy-pasted. Checking test coverage would be good though.

[the remark this answers is not in the document]
Comment (2LoS) [reply libcxx/include/forward_list:557]: @EricWF, could you help me to land this PR, please?

**Review by philnik777 — APPROVED** [review]
LGTM assuming we have (indirect) test coverage for `__delete_node`.

**Comment by 2LoS** [comment]
@philnik777, can you land this for me, please?

**Pull request merged**